### PR TITLE
ci: Upload binaries to github release

### PR DESCRIPTION
**Pull request by soluchok**
Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -49,3 +49,13 @@ jobs:
...
             ${{ env.IMAGE_PREFIX }}:${{ env.IMAGE_TAG }}
             ${{ env.IMAGE_PREFIX }}:latest
+
+      - name: Build binaries
+        run: make build-vct
+
+      - name: Upload binaries and their SHA to Github Release
+        uses: alexellis/upload-assets@0.3.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (soluchok) [inline .github/workflows/publish.yaml:59]: @troyronda token needs to be set

File: Makefile [hunk 2]
```diff
@@ -40,7 +40,10 @@ bdd-test: build-vct-docker
 .PHONY: build-vct
 build-vct:
```
Comment (troyronda) [inline Makefile:41]: Note: this is build VCT for distribution.

File: Makefile [hunk 3]
```diff
@@ -40,7 +40,10 @@ bdd-test: build-vct-docker
...
 	@echo "Building verifiable credentials transparency (vct)"
-	@go build -o build/bin/vct cmd/vct/main.go
+	@CGO_ENABLED=0 GOOS=linux go build -o build/dist/bin/vct-linux cmd/vct/main.go
```
Comment (troyronda) [inline Makefile:43]: This is the amd64 linux build. Should explicitly add GOARCH=amd64 and include in the binary name.

File: Makefile [hunk 4]
```diff
@@ -40,7 +40,10 @@ bdd-test: build-vct-docker
...
+	@CGO_ENABLED=0 GOOS=darwin go build -o build/dist/bin/vct-darwin cmd/vct/main.go
```
Comment (troyronda) [inline Makefile:44]: This is the amd64 darwin build. Should explicitly add GOARCH=amd64 and include in the binary name.

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (troyronda) [reply .github/workflows/publish.yaml:59]: @soluchok https://docs.github.com/en/actions/reference/authentication-in-a-workflow

File: Makefile (hunk 3 above)
Comment (troyronda) [reply Makefile:43]: Also arm64 is also useful.

File: Makefile (hunk 4 above)
Comment (troyronda) [reply Makefile:44]: Also arm64 is also important for darwin.

File: Makefile [hunk 5]
```diff
@@ -40,7 +40,10 @@ bdd-test: build-vct-docker
...
+	@CGO_ENABLED=0 GOOS=windows GOARCH=amd64 go build -o build/dist/bin/vct-amd64.exe cmd/vct/main.go
```
Comment (troyronda) [inline Makefile:45]: Not sure we are really supporting windows for non-cli scenarios.

File: Makefile [hunk 6]
```diff
@@ -42,6 +42,15 @@ build-vct:
 	@echo "Building verifiable credentials transparency (vct)"
 	@go build -o build/bin/vct cmd/vct/main.go
 
+.PHONY: build-vct-dist
+build-vct-dist:
+	@echo "Building verifiable credentials transparency (vct)"
```
Comment (troyronda) [inline Makefile:47]: TODO: determine if there is a separate binary for the CLI.

**Review by troyronda — APPROVED** [review]

**Pull request merged**